src/hooks/users.ts

```diff
@@ -1,15 +1,97 @@
 import { useMemo } from "react";
-import { isLoaded } from "react-redux-firebase";
+import { isLoaded, useFirestoreConnect } from "react-redux-firebase";
 
-import { usersSelector, usersByIdSelector } from "utils/selectors";
+import { universeUsersSelector, usersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
 import { useSelector } from "hooks/useSelector";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 import { useVenueId } from "hooks/useVenueId";
 import { useSparkleFirestoreConnect } from "hooks/useSparkleFirestoreConnect";
+import { useConnectRelatedVenues } from "hooks/useConnectRelatedVenues";
 
 import { User } from "types/User";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
+
+export const useRecentPartyUsers = () => [];
+// export const useRecentPartyUsers = (venueName: string): WithId<User>[] => {
+//   const venueId = useVenueId();
+//   const partygoers = useSelector(partygoersSelector) ?? [];
+
+//   const { parentVenue, relatedVenues } = useConnectRelatedVenues({
+//     venueId,
+//   });
+
+//   const lastSeenThreshold = useUserLastSeenThreshold();
+
+//   const parentVenueName = parentVenue?.name ?? "";
+
+//   const filterPartygoersByHierarchy = useCallback(
+//     (partygoer: WithId<User>) => {
+//       return relatedVenues?.map((venue) => {
+//         return (
+//           partygoer?.lastSeenIn?.[parentVenueName || venueName || venue.name] >
+//           lastSeenThreshold
+//         );
+//       });
+//     },
+//     [lastSeenThreshold, parentVenueName, relatedVenues, venueName]
+//   );
+
+//   return useMemo(() => partygoers.filter(filterPartygoersByHierarchy), [
+//     partygoers,
+//     filterPartygoersByHierarchy,
+//   ]);
+// };
+
+// NOTE: Heuristic approach. Way faster and efficient.
+// Is based on assumption that there is only one parent venue and it is an entry point for any user
+export const useConnectUniverseUsers = () => {
```

Comment:
The following is off the top of my head/from what I remember, I haven't deeply checked/validated it against the codebase to confirm:

- I believe that in the case of new users, when there is only a single level of venue 'depth', they should always end up entering through the 'main' venue (which is likely what has been used for the `parentVenueId`)
- If they are an existing user that has logged in via another event/is already logged in, and then navigate directly to a sub-venue, then they won't be counted as having entered the 'main' venue. So this assumption won't be true.
- If there are multiple 'levels' of subvenue (eg. `A -> B -> C -> D`) then I believe the `parentVenueId` of the 'deeper' children (eg. `C`, `D`) won't be set to the 'main' venue (eg. `A`). Only the main venue's direct children (eg. `B`) will have the `parentVenueId` set to the 'main' venue'.